types/aws-lambda/trigger/api-gateway-proxy.d.ts

```diff
@@ -4,9 +4,32 @@ import {
 } from "../common/api-gateway";
 import { Callback, Handler } from "../handler";
 
+/**
+ * Works with Lambda Proxy Integration for Rest API or HTTP API integration Payload Format version 1.0
+ * @see - https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html
+ */
 export type APIGatewayProxyHandler = Handler<APIGatewayProxyEvent, APIGatewayProxyResult>;
+/**
+ * Works with Lambda Proxy Integration for Rest API or HTTP API integration Payload Format version 1.0
+ * @see - https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html
+ */
 export type APIGatewayProxyCallback = Callback<APIGatewayProxyResult>;
 
+/**
+ * Works with HTTP API integration Payload Format version 2.0
+ * @see - https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html
+ */
+export type APIGatewayProxyHandlerV2 = Handler<APIGatewayProxyEventV2, APIGatewayProxyResultV2>;
+/**
+ * Works with HTTP API integration Payload Format version 2.0
+ * @see - https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html
+ */
+export type APIGatewayProxyCallbackV2 = Callback<APIGatewayProxyResultV2>;
+
+/**
+ * Works with Lambda Proxy Integration for Rest API or HTTP API integration Payload Format version 1.0
+ * @see - https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html
+ */
```

Comment:
👍 on doc comments, it would be great to get better coverage on this, given how broken up all the AWS documentation is, so adding it on the go is nice, though at least having "Event for ..." etc vs "Works with" would read a little less weird in isolation.

Reply:
What's your suggestion?

Reply:
As mentioned, replace the generic "Works with" with "Event for" / "Result for" etc., which has the same information, but the paragraph is now a description of the documented type, which is what doc comments are for.